### PR TITLE
Liquidsoap daemon 0.3.0

### PR DESCRIPTION
**Pull request by toots**
This release does not try to run the daemon when installing..

### CODE REVIEW

**Comment by toots** [comment]
Unfortunately, I think that calling `sudo` is the problem.

**Comment by avsm** [comment]
still appears to be trying to run it...

```
Command started
+ ./daemonize-liquidsoap.sh  (CWD=/home/opam/.opam/4.03.0/build/liquidsoap-daemon.0.3.0)
- Failed to get D-Bus connection: Unknown error -1
[ERROR] The compilation of liquidsoap-daemon failed at
        "./daemonize-liquidsoap.sh".
[liquidsoap-daemon: env] Command started
+ env "mode=remove" "./daemonize-liquidsoap.sh" (CWD=/home/opam/.opam/4.03.0/build/liquidsoap-daemon.0.3.0)
- Failed to get D-Bus connection: Unknown error -1
- Failed to get D-Bus connection: Unknown error -1

```

**Comment by avsm** [comment]
I'll close this one to reduce pressure on the CI backlog -- please do reopen if there is a fix to the daemon starting issue!

**Comment by toots** [comment]
Hi @avsm and thanks for looking at this once more.

This release does not start the daemon, I believe that the D-Bus issues comes from trying to call `sudo`.

I'm a little hesitant as to how to push this further as this package isn't really an ocaml program or module.

Is there a defined policy from the opam managers as to what falls into the scope of an opam package and what opam installs are supposed to do or not do?

File: packages/liquidsoap-daemon/liquidsoap-daemon.0.3.0/opam [hunk 1]
```diff
@@ -0,0 +1,22 @@
+opam-version: "1.2"
+maintainer: "romain.beauxis@gmail.com"
+homepage: "https://github.com/savonet/liquidsoap-daemon"
+authors: "The Savonet Team <savonet-users@lists.sourceforge.net>"
+available: [ os != "win32" ]
+install: [
+  ["./daemonize-liquidsoap.sh"] { os = "linux" }
```
Comment (avsm) [inline packages/liquidsoap-daemon/liquidsoap-daemon.0.3.0/opam]: this script is called `daemonize-liquidsoap` -- is it really meant to be in the install field?

Comment (toots) [inline packages/liquidsoap-daemon/liquidsoap-daemon.0.3.0/opam]: The name may be misleading. All it does now is install the daemonization script/configuration.

**Comment by avsm** [comment]
> This release does not start the daemon, I believe that the D-Bus issues comes from trying to call sudo.

Are you sure that the daemonization fix was pushed correctly? The diff above still has a script called `daemonize-liquidsoap`, hence my query :-)

> Is there a defined policy from the opam managers as to what falls into the scope of an opam package and what opam installs are supposed to do or not do?

In OPAM1, in general any unprivileged operations within `~/.opam` are ok. In OPAM2 (still in beta), it is much stricter about tracking activities and files installed, but that is still under development and not a blocker to merging to this package repository.  In this case, the CI is failing due to the installation script (which should be unprivileged), doing something that requires root on Linux.  I've not had a chance to dig into why that is beyond looking at the logs and seeing the dbus errors.

File: packages/liquidsoap-daemon/liquidsoap-daemon.0.3.0/opam (hunk 1 above)
Comment (avsm) [inline packages/liquidsoap-daemon/liquidsoap-daemon.0.3.0/opam]: Looking at the script, it appears to call sudo for removal as well

```
            sudo systemctl disable liquidsoap
            sudo systemctl stop liquidsoap
            sudo rm "$systemd_target"
            sudo systemctl daemon-reload
```

I think any invocation of sudo like this as part of the opam install would be the problem -- we don't really have a good way to hook into system daemon scripts in a portable way, so the best way is to print a `messages:` field asking the user to run the script manually via sudo.  @altgr may have a better idea, so CCIng him

**Comment by toots** [comment]
Indeed, the original intent was to have system-wide, privileged files installed. I'll review and change that.

Thanks for the feedback and review!

**Comment by avsm** [comment]
No problem - the issue you raised has been brought up by other 'daemon'-type applications in OPAM such as xapi (cc @jonludlam) and it's never been clear how to package them systemwide.

I think the probable answer is that deb/rpm/docker packaging takes care of that, and OPAM packages should perhaps install local configuration files and not hook into systemd/launchd themselves.

**Comment by toots** [comment]
Yeah, that's what I plan on doing.

**Pull request closed without merging**